src/platform.ts

```diff
@@ -70,16 +70,38 @@ export function getDefaultPowerShellPath(
     // Find the path to powershell.exe based on the current platform
     // and the user's desire to run the x86 version of PowerShell
     if (platformDetails.operatingSystem === OperatingSystem.Windows) {
-        if (use32Bit) {
-            powerShellExePath =
-                platformDetails.isOS64Bit && platformDetails.isProcess64Bit
-                    ? SysWow64PowerShellPath
-                    : System32PowerShellPath;
+        const psCoreInstallPath =
+        (!platformDetails.isProcess64Bit ? process.env.ProgramW6432 : process.env.ProgramFiles) + "\\PowerShell";
```

Comment:
Does `process.env.ProgramW6432` resolve properly on 32 bit Windows?

Reply:
Not sure what you mean @corbob

Reply:
What I mean is that on 32 bit Windows, the environment variable `ProgramW6432` doesn't exist... So if someone is on 32 bit Windows with PowerShell Core installed (my first question is why?!?!?!?!) runs, it won't default to Core.

That being said: if I were to guess, I would say the number of people using vs code on 32 bit windows with PowerShell Core is probably really close to 0...

Reply:
That's odd. What _is_ the 32bit program files location/is there an env var that gives us that?

Reply:
on 32bit Windows it's just `ProgramFiles` and it resolves to C:\Program Files (usually)

I think if you just check for `ProgramFiles`, then the PowerShell will match the bitness of the process if it can find it...

Reply:
Ah ok

So what we want to do is:

If 32 bit, check ProgramFiles
If 64 bit, check both ProgramFiles and ProgramW6432

?

Reply:
Something like this? 
```
    if (platformDetails.operatingSystem === OperatingSystem.Windows) {
        if (use32Bit){
            psCoreInstallPath = process.env.ProgramFiles + "\\PowerShell"
        }
        else {
            psCoreInstallPath =
            (platformDetails.isProcess64Bit ? process.env.ProgramFiles : process.env.ProgramW6432) + "\\PowerShell";
        }
```

Reply:
Not quite...

```typescript
    if (platformDetails.operatingSystem === OperatingSystem.Windows) {
        if (use32Bit) {
           psCoreInstallPath =  (platformDetails.isProcess64Bit ? process.env.ProgramFiles(x86) : process.env.ProgramFiles) + "\\PowerShell";
           
        } else {
            psCoreInstallPath = (platformDetails.isProcess64Bit ? process.env.ProgramFiles : process.env.ProgramW6432) + "\\PowerShell";;
        }
```

The Program Files environment variables on my computer (in 64 bit PowerShell process):

```
ProgramFiles                   C:\Program Files // 64 bit Program Files
ProgramFiles(x86)              C:\Program Files (x86) // 32 bit Program Files
ProgramW6432                   C:\Program Files // 64 bit Program Files
```

Which might bring up a concern of the `(` and `)` in the variable name... might be worth pausing the debugger to inspect `process.env` to see what environment variables it has?

Reply:
Not going to edit it, will just leave it here. To access the x86 Program Files Environment Variable, you seem to need to do `process.env['ProgramFiles(x86)']`